Ginger/Ginger/ALM/JIRA/JiraZephyrCyclesExplorerPage.xaml.cs

```diff
@@ -0,0 +1,259 @@
+using GingerCore;
+using System;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using GingerCore.ALM;
+using Ginger.ALM.JIRA.TreeViewItems;
+using amdocs.ginger.GingerCoreNET;
+using JiraRepository.Data_Contracts;
+using GingerCore.ALM.JIRA;
+
+namespace Ginger.ALM.JIRA
+{
+    public partial class JiraZephyrCyclesExplorerPage : Page
+    {
+        public enum eExplorerTestLabPageUsageType { Import, Select, BrowseFolders }
+
+        ObservableList<BusinessFlow> mBizFlows;
+        private List<string[]> mExecDetailNames;
+        JiraZephyrCyclesCollection treeData;
+
+        private ITreeViewItem mCurrentSelectedTreeItem = null;
+        public string CurrentSelectedPath { get; set; }
+        ObservableList<JiraZephyrVersionTreeItem> mCurrentSelectedVersions = new ObservableList<JiraZephyrVersionTreeItem>();
+        ObservableList<JiraZephyrCycleTreeItem> mCurrentSelectedCycles = new ObservableList<JiraZephyrCycleTreeItem>();
+        object mCurrentSelectedObject = new object();
+        private string mImportDestinationPath = string.Empty;
+        private bool mParentSelectionMode = false;
+        GenericWindow _GenericWin = null;
+
+        public ObservableList<JiraZephyrVersionTreeItem> CurrentSelectedVersions
+        {
+            get
+            {
+                return mCurrentSelectedVersions;
+            }
+            set
+            {
+                mCurrentSelectedVersions = value;
+            }
+        }
+
+        public ObservableList<JiraZephyrCycleTreeItem> CurrentSelectedCycles
+        {
+            get
+            {
+                return mCurrentSelectedCycles;
+            }
+            set
+            {
+                mCurrentSelectedCycles = value;
+            }
+        }
+
+        public JiraZephyrCyclesExplorerPage(string importDestinationPath = "", bool parentSelectionMode = false)
+        {
+            InitializeComponent();
+
+            mImportDestinationPath = importDestinationPath;
+            mParentSelectionMode = parentSelectionMode;
+
+            GetTreeData();
+
+            JiraZephyrCyclesExplorerTreeView.TreeTitle = "'" + ALMCore.DefaultAlmConfig.ALMDomain + " \\ " + ALMCore.DefaultAlmConfig.ALMProjectName + "' - Test Cycles Explorer";
+            JiraZephyrCyclesExplorerTreeView.TreeTitleStyle = (Style)TryFindResource("@ucTitleStyle_3");
+            JiraZephyrCyclesExplorerTreeView.Tree.ItemSelected += TestLabExplorerTreeView_ItemSelected;
+            foreach (JiraZephyrRelease version in treeData.projectsReleasesList)
+            {
+                JiraZephyrVersionTreeItem tvv = new JiraZephyrVersionTreeItem(version.releasesCycles);
+                tvv.Name = version.versionName.ToString();
+                JiraZephyrCyclesExplorerTreeView.Tree.AddItem(tvv);
+            }
+
+            LoadDataBizFlows();
+
+            // mExecDetailNames = new List<string[]>();
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6280018)